app/recipe_miner/queries/recipe_with_ingredients.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+module Queries
+  class RecipeWithIngredients
+    def initialize(ingredients:)
+      @ingredients = ingredients
+    end
+
+    def result
```

Comment:
I'd be aware/wary of classes with one method that takes no arguments. To me it's a sign that either the method should take arguments, or that the whole thing should be part of another class (maybe both).

To explain why the method should take arguments, I think a simple reason is that it becomes more flexible. We can pass around a reusable object and call it multiple times with different args, rather than having to throw it away after one use.

If we forget about the current names of the class/method we would have:

```ruby
class Foo
  def bar(ingredients)
    ...
  end
end
```

which says that a `Foo` object has a method `bar` which returns a single recipe (or throws) given a list of ingredients. From that I think it's clear that `Foo` is `Recipe` and `bar` is a scope on it, so maybe:

```ruby
class Recipe
  def self.with_ingredients(ingredients)
    ...
  end
end
```

Here the object is really the class `Recipe` itself which we can pass around as if it were an object, so `Recipe.with_ingredients(ingredients)`.

Reply:
On the other hand, let's say that `RecipeWithIngredients` somehow needs an external service, like S3. Then you'd probably want to inject the S3 client, but in a way that doesn't expose it to the outside.

In that case you may have:

```
class RecipeWithIngredients
  def initialise(s3_client)
    ...
  end
  def results(ingredients)
    # do something with s3
  end
end
```

which would be totally fine too, but starting to slightly go against Rails conventions. As far as I know Rails doesn't have the ability to do dependency injection like this out of the box.